### PR TITLE
Fix propTypes warnings on client side render

### LINKED ISSUE #4285: propTypes warnings on client side render
**Describe the bug**
`npm run dev`, visit http://localhost.bbc.com:7080/news/articles/c0g992jmmkko and click the inline link (any link inside the article main paragraphs), see errors in console.

**Expected behavior**
There should not be propTypes errors in the console when performing an onward journey using the client app.

**Screenshots**
image

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by stevenwally**
Resolves #4285 

**Overall change:** _On subsequent page renders (eg. clicking a link in an article), we want to set the status code to be a meaningful value so that child components (that require this value) can make use of it, instead of unconditionally setting to `null`. When tested, clicking links within articles should no longer produce warnings about required `propTypes`_

**Code changes:**
_In state,_ `status` _(status code) was being unconditionally set to `null` on every subsequent render. There are multiple components that, per their propType definitions, require the `status` prop and thus it cannot be `null` when passed to these components without raising warnings from React_

_The chosen approach was to, when setting state, to assign the value of `status` (from state). If `status` doesn't exist, continue to use `null` as a fallback._

_(As an alternative approach, the affected component propTypes could simply not explicitly require `status`/`statusCode`. Happy to make that change if it is more desirable.)_

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by dr3 — COMMENTED** [review]
Oops, In this case I actually think the proptypes are wrong on `latest`, and that it shouldn't be required 🙈 , when the app goes into `loading` state we clear the `pageData`, `status` etc and set `loading` to `true`.

Anything between `simorgh/src/app/containers/App/App.jsx` and `simorgh/src/app/containers/PageHandlers/withLoading` in the container order should allow `pageData` and `status` to be optional I think 🙈 This thus includes `WithContextsContainer`, `PageWrapperContainer` & `RequestContextProvider`. The order can be seen [here](https://github.com/bbc/simorgh/blob/attempt-bundle-splitting-for-CPS-asset_signed-commits/src/app/containers/Article/index.jsx#L26-L28) in the order theyre loaded and nested :)

I see from your PR body youve noticed this is an option too, if you could switch to this we would be very grateful :)

**Comment by stevenwally** [comment]
Ah, got it! Thanks a bunch, @dr3 :) I'll revert the original changes that I made and update the prop type requirements for those components. 

Just as an aside, is there a reason you wouldn't want to set the status code to something like `100` when loading? (This is _by all means_ more of a curiosity than a suggestion 😄 was just musing about it and wondered if `100` [could be more meaningful](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/100) than a `null`)

**Comment by dr3** [comment]
> Just as an aside, is there a reason you wouldn't want to set the status code to something like `100` when loading? (This is _by all means_ more of a curiosity than a suggestion 😄 was just musing about it and wondered if `100` [could be more meaningful](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/100) than a `null`)

Ah interesting, weve never thought of that, as it stands we only really use the `loading` value during that time, so im not sure it would do much, but def something we can think about :)

File: .env [hunk 1]
```diff
@@ -1,4 +1,8 @@
-# On running the 'build' commands this file is replaced by the files in `env-config/`
-# To ignore and keep this file we have a command in `postshrinkwrap` which assumes this file is not changing
-# The command is `git update-index --assume-unchanged .env` 
-# Source: https://wildlyinaccurate.com/git-ignore-changes-in-already-tracked-files/
+SIMORGH_BASE_URL=http://localhost.bbc.com:7080
```
Comment (dr3) [inline .env]: This file needs resetting, it appears `git update-index --assume-unchanged .env` didnt run on your machine right 🙈 If you do `git checkout latest .env` commit & push it, and then do `git update-index --assume-unchanged .env` it should stop it happening again :)

Comment (stevenwally) [reply .env]: Ah! Whoops 😅 thanks!

**Review by dr3 — APPROVED** [review]
LGTM, thanks for the contribution :D 

Travis is failing due to known vulnerability that were aiming to resolve on monday :)

**Review by jroebu14 — APPROVED** [review]

**Pull request merged**